contracts/cairo/src/escrow.cairo

```diff
@@ -117,15 +117,17 @@ mod Escrow {
         fn set_order(ref self: ContractState, order: Order) -> u256 {
             assert(order.amount > 0, 'Amount must be greater than 0');
 
+            let payment_amount = order.amount + order.fee;
+            let dispatcher = IERC20Dispatcher { contract_address: self.native_token_eth_starknet.read() };
+            assert(dispatcher.allowance(get_caller_address(), get_contract_address()) >= payment_amount, 'Not enough allowance');
```

Comment:
I have a question, is it possible to have an allowance without having a balance? Because if that's the case, maybe it's advisable to specify that both the allowance and the balance should be >= `payment_amount`